core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -553,8 +553,23 @@ private void runInternal() {
               context.cancel(null);
               return;
             }
-            listener =
-                startCall(stream, methodName, method, headers, context, statsTraceCtx, tag);
+            listener = startCall(stream, methodName, method, headers, context, statsTraceCtx, tag);
+
+            // Delay cancellation to after startCall().
```

Comment:
I don't think this is right. It just needs to be after the `stream.setListener()` (which means also after jumpListener.setListener() which is described in the comment above).

I will say this comment isn't all that helpful as-is because it doesn't explain _why_ delaying is important.